apps/findmydevice/js/findmydevice.js

```diff
@@ -208,26 +244,31 @@ var FindMyDevice = {
   },
 
   _contactServer: function fmd_contact_server() {
+    var p;
     this.beginHighPriority('clientLogic');
     if (this._registered && this._enabled) {
-      this._replyAndFetchCommands();
+      p = this._replyAndFetchCommands();
     } else if (this._registered && !this._enabled) {
-      this._reportDisabled();
+      p = this._reportDisabled();
     } else if (!this._registered && this._enabled) {
-      this._register();
+      p = this._register();
     } else {
       // XXX(ggp) this should never happen, but let's play
       // it safe and release lock we acquired
       DUMP('can\'t contact the server while not registered and not enabled!!');
-      this.endHighPriority('clientLogic');
+      p = Promise.reject();
     }
+
+    p.then(
```

Comment:
So I suspect we want to release the wake lock in all cases, and that's why we have those two calls ?